src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -0,0 +1,68 @@
+package com.mathworks.ci;
+/**
+ * Copyright 2019-2020 The MathWorks, Inc.
+ * 
+ * Build Interface has two default methods. MATLAB builders can override the 
+ * default behavior.
+ * 
+ */
+
+import java.io.IOException;
+import java.io.InputStream;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Launcher.ProcStarter;
+import hudson.model.TaskListener;
+
+public interface MatlabBuild {
+    
+    /**
+     * This Method decorates the launcher with MATLAB command provided and returns the Process 
+     * object to launch MATLAB with appropriate startup options like -r or -batch
+     * @param workspace
+     * @param launcher
+     * @param listener
+     * @param envVars
+     * @param matlabCommand
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    default ProcStarter getProcessToRunMatlabCommand(FilePath workspace, Launcher launcher,TaskListener listener, EnvVars envVars, String matlabCommand) throws IOException, InterruptedException {
+        ProcStarter matlabLauncher;

```

Comment:
Is this Java convention to declare first and then assign? Or could this just be
ProcStarter matlabLauncher = launher.launch().....

Reply:
the matlabLauncher is updated based on the branching condition for either Linux/Windows hence i choose to keep it out of the branching scope and return it after branching is done.

Reply:
Made appropriate changes